generators/server/templates/src/main/java/package/config/apidoc/SwaggerSecurityCustomizer.java.ejs

```diff
@@ -0,0 +1,96 @@
+<%#
+ Copyright 2013-2019 the original author or authors from the JHipster project.
+
+ This file is part of the JHipster project, see https://www.jhipster.tech/
+ for more information.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-%>
+package <%=packageName%>.config.apidoc;
+
+import com.google.common.base.Predicate;
+import com.google.common.base.Predicates;
+import io.github.jhipster.config.apidoc.customizer.SwaggerCustomizer;
+import org.springframework.stereotype.Component;
+import springfox.documentation.builders.PathSelectors;
+import springfox.documentation.service.ApiKey;
+import springfox.documentation.service.AuthorizationScope;
+import springfox.documentation.service.SecurityReference;
+import springfox.documentation.service.SecurityScheme;
+import springfox.documentation.spi.service.contexts.SecurityContext;
+import springfox.documentation.spring.web.plugins.Docket;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static com.google.common.collect.Lists.newArrayList;
+import static springfox.documentation.builders.PathSelectors.regex;
+
+@Component
+public class SwaggerSecurityCustomizer implements SwaggerCustomizer {
+<%_ if (applicationType === 'monolith' || applicationType === 'gateway') { _%>
+
+    private static final String[] PUBLIC_ENDPOINTS = {
```

Comment:
It could be good to have the same list in SecurityConfiguration and here. I just tried to move this list in SecurityConfiguration and that's pretty nice:

Before:
```java
.antMatchers("/api/authenticate").permitAll()
.antMatchers("/api/register").permitAll()
.antMatchers("/api/activate").permitAll()
.antMatchers("/api/account/reset-password/init").permitAll()
.antMatchers("/api/account/reset-password/finish").permitAll()
```

After:
```java
.antMatchers(PUBLIC_ENDPOINTS).permitAll()
```

Then you could reuse this list here:
```java
.securityContexts(newArrayList(new SecurityContext(bearerAuth(), not(SecurityConfiguration.PUBLIC_ENDPOINTS))));
```

What do you think?

Reply:
Looks good to me. I was also looking for a way to get these endpoints programmatically. I wanted to avoid modifying the security configuration. but I couldn't find a solution.

Reply:
I had to remove the customizer to avoid introducing breaking changes. We will reconsider this for v7